### ISSUE TITLE
Strict validation of input samplesheet.csv

### ISSUE THREAD

**Issue by jannikseidelQBiC** [nextflow-io/nf-schema#7]
Following feature would be beneficial in the future:

- The headers of the samplesheet should be able to be validated in `strict` mode, meaning if they are misspelled or missing the pipelines should throw an error instead of a warning
- This behavior should be able to be turned on or off

The same feature would be nice for entries in the samplesheet, where required fields are missing.

e.g. if following samplesheet would be present

``` csv
sample,short_reads_fastq_1,short_reads_fastq_2,long_reads_fastq_2
test,,path/to/first_fastq.gz,
```
and should be formatted like this 

``` csv
sample,short_reads_fastq_1,short_reads_fastq_2,long_reads_fastq_1
test,path/to/first_fastq.gz,,
```
and either `short_reads_fastq_1` or `long_reads_fastq_1`are required fields

The pipeline should fail and not run empty with warnings. Also a meaningful error should be thrown in such a scenario.

It would be best this could be turned on or off in a [schema_input.json](https://raw.githubusercontent.com/nf-core/detaxizer/dev/assets/schema_input.json).

**Comment by nvnieuwk** [comment]
The best solution for this is to create a new configuration option (`validation.strictHeaders` for example). This can then be turned on to throw errors instead of warnings for unknown headers.
